### PR TITLE
JavaKit: add CustomJavaClassLoaderConstructible protocol

### PR DESCRIPTION
**Pull request by lhoward**
Swift projections of Java classes can conform to
CustomJavaClassLoaderConstructible, to provide a custom class loader to be used when constructing new instances.

Fixes: #154

### CODE REVIEW

File: Tests/Java2SwiftTests/Java2SwiftTests.swift [hunk 1]
```diff
@@ -12,7 +12,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-import JavaKit
+@testable import JavaKit
```
Comment (DougGregor) [inline Tests/Java2SwiftTests/Java2SwiftTests.swift]: Could we use `@_spi(Testing)` here instead of `@testable`?

**Review by DougGregor — COMMENTED** [review]
This is a great addition, thank you!

**Comment by DougGregor** [comment]
> Bikeshed: `CustomJavaClassLoaderConstructible` or `CustomJavaClassLoaderInitializable`?

I'd prefer to avoid `Constructible`, since that's not really a thing in Swift. How about just `CustomJavaClassLoader`, as in, this type has a custom Java class loader?

**Comment by DougGregor** [comment]
@lhoward anything else you want to do here before we merge?

**Comment by lhoward** [comment]
> @lhoward anything else you want to do here before we merge?

I'm happy for you to merge, unless you want tests (which I am still working on).

**Comment by DougGregor** [comment]
I do like tests. Let's wait until you have some tests in place.

**Comment by lhoward** [comment]
I'm a bit caught up with other stuff today, I'll endeavour to get onto tests next week, feel free to merge (or not) in the interim.

**Comment by DougGregor** [comment]
Since this is also needed in the other PR, I'll go ahead and merge.

**Pull request merged**